### PR TITLE
add `D.Exponential`

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**

### CODE REVIEW

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

File: chainer/distributions/exponential.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    @property
+    def lam(self):
+        return self.__lam
+
+    @property
+    def batch_shape(self):
+        return self.lam.shape
+
+    def cdf(self, x):
+        return 1 - exponential.exp(-self.lam * x)
```
Comment (toslunar) [inline chainer/distributions/exponential.py]: Could you use `expm1`?

File: chainer/distributions/exponential.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    @property
+    def entropy(self):
+        return 1 - exponential.log(self.lam)
+
+    @property
+    def event_shape(self):
+        return ()
+
+    def icdf(self, x):
+        return -1 / self.lam * exponential.log(1 - x)
```
Comment (toslunar) [inline chainer/distributions/exponential.py]: Could you use `log1p`?

File: tests/chainer_tests/distributions_tests/test_exponential.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+from chainer import distributions
+from chainer import testing
+import numpy
```
Comment (toslunar) [inline tests/chainer_tests/distributions_tests/test_exponential.py]: Could you fix the grouping of imports?

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

File: tests/chainer_tests/distributions_tests/test_exponential.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    scipy_onebyone = True
+
+    def setUp_configure(self):
+        from scipy import stats
+        self.dist = distributions.Exponential
+        self.scipy_dist = stats.expon
+
+        self.test_targets = set([
+            "batch_shape", "cdf", "entropy", "event_shape", "log_prob",
+            "mean", "sample", "support", "variance"])
```
Comment (toslunar) [inline tests/chainer_tests/distributions_tests/test_exponential.py]: `icdf` is defined but not tested.

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Pull request merged**